Kconfig.zephyr

```diff
@@ -363,6 +363,12 @@ config LEGACY_DEVICETREE_MACROS
 	  Zephyr 2.2 and previous versions, rather than the devicetree.h
 	  API introduced during the Zephyr 2.3 development cycle.
 
+config DEPRECATED_ZEPHYR_INT_TYPES
```

Comment:
I think we should use `LEGACY`, like we've done for DT and timeouts

Reply:
```suggestion
config LEGACY_ZEPHYR_INT_TYPES
```

Reply:
Legacy means that you want to maintain this beyond the deprecation time, deprecated is more appropriate IMO, meaning at some point it is going away.